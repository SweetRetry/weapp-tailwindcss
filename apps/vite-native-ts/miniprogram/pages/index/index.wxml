<view class="p-4">
  <IceButton class="bg-[#d72929]" my-class="bg-[orange]"></IceButton>
</view>