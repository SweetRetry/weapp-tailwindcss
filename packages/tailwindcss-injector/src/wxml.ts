import type { Buffer } from 'node:buffer'
import path from 'node:path'
import fs from 'fs-extra'
import { Parser } from 'htmlparser2'
import MagicString from 'magic-string'

const srcImportTagsMap: Record<string, string[]> = {
  // audio: ['src', 'poster'],
  // video: ['src', 'poster'],
  // image: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxs/01wxs-module.html
  // wxs: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxml/import.html
  import: ['src'],
  include: ['src'],
}

export interface WxmlDep {
  tagName: string
  start: number
  end: number
  quote: string | null | undefined
  name: string
  value: string
  attrs: Record<string, string>
}

// https://github.com/fb55/htmlparser2/issues/1541
// https://developers.weixin.qq.com/miniprogram/dev/framework/view/wxml/event.html
export function processWxml(wxml: string | Buffer) {
  const ms = new MagicString(wxml.toString())
  const deps: WxmlDep[] = []
  let currentTagName = ''
  let importAttrs: undefined | string[]
  let attrs: Record<string, string> = {}

  const parser = new Parser(
    {
      onopentagname(name) {
        currentTagName = name
        importAttrs = srcImportTagsMap[currentTagName]
      },
      onattribute(name, value, quote) {
        attrs[name] = value
        if (importAttrs) {
          for (const attrName of importAttrs) {
            if (attrName === name) {
              deps.push({
                name,
                value,
                quote,
                tagName: currentTagName,
                start: parser.startIndex,
                end: parser.endIndex,
                attrs,
              })
            }
          }
        }
      },
      onclosetag() {
        currentTagName = ''
        attrs = {}
        importAttrs = undefined
      },
    },
    {
      lowerCaseTags: false,
      xmlMode: true,
    },
  )
  parser.write(
    ms.original,
  )
  parser.end()
  return deps
}

export async function getDepFiles(file: string) {
  const set = new Set<string>()

  async function getDep(filepath: string) {
    if (await fs.exists(filepath)) {
      const dirname = path.dirname(filepath)
      set.add(filepath)
      const wxml = await fs.readFile(filepath, 'utf8')
      const deps = await processWxml(wxml)
      for (const { value } of deps) {
        const p = path.resolve(dirname, value)
        if (!set.has(p)) {
          await getDep(p)
        }
      }
    }
  }

  await getDep(file)

  return set
}
