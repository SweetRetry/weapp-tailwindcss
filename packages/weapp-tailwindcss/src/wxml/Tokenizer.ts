// 针对微信的属性表达式进行切词
// https://github.com/sonofmagic/weapp-tailwindcss/issues/319
// all state
export enum State {
  START,
  TEXT,
  OPEN_BRACE,
  POTENTIAL_CLOSE,
  BRACES_COMPLETE,
}

export class TokenizerStateMachine {
  private state: State
  private buffer: string
  private tokens: string[]

  constructor() {
    this.state = State.START
    this.buffer = ''
    this.tokens = []
  }

  private processChar(char: string) {
    switch (this.state) {
      case State.START:
        if (char === ' ') {
          // Ignore leading spaces
        }
        else if (char === '{') {
          this.state = State.OPEN_BRACE
          this.buffer += char
        }
        else {
          this.state = State.TEXT
          this.buffer += char
        }
        break

      case State.TEXT:
        if (char === ' ') {
          this.tokens.push(this.buffer)
          this.buffer = ''
          this.state = State.START
        }
        else if (char === '{') {
          this.buffer += char
          this.state = State.OPEN_BRACE
        }
        else {
          this.buffer += char
        }
        break

      case State.OPEN_BRACE:
        if (char === '}') {
          this.buffer += char
          this.state = State.POTENTIAL_CLOSE
        }
        else {
          this.buffer += char
        }
        break

      case State.POTENTIAL_CLOSE:
        if (char === '}') {
          this.buffer += char
          this.state = State.BRACES_COMPLETE
        }
        else {
          this.buffer += `}${char}`
          this.state = State.OPEN_BRACE
        }
        break

      case State.BRACES_COMPLETE:
        if (char === ' ') {
          this.tokens.push(this.buffer)
          this.buffer = ''
          this.state = State.START
        }
        else {
          this.buffer += char
          this.state = State.TEXT
        }
        break

      default:
        throw new Error('Unexpected state')
    }
  }

  public run(input: string): string[] {
    for (const char of input) {
      this.processChar(char)
    }
    // Push the last buffer if it's not empty
    if (this.buffer.length > 0) {
      this.tokens.push(this.buffer)
    }
    return this.tokens
  }
}
