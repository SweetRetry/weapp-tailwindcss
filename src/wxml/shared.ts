import { MappingChars2String as dic } from '@/dic'
import { ICommonReplaceOptions } from '@/types'
import { mangleMark } from '@/mangle/expose'

// export function arbitraryValuesReplacer (str: string) {}

export function replaceWxml (
  original: string,
  options: ICommonReplaceOptions | boolean = {
    keepEOL: false,
    mangle: false
  }
) {
  if (typeof options === 'boolean') {
    options = {
      keepEOL: options,
      mangle: false
    }
  }
  let res = original
    .replace(/\[/g, dic['['])
    .replace(/\]/g, dic[']'])
    .replace(/\(/g, dic['('])
    .replace(/\)/g, dic[')'])
    .replace(/#/g, dic['#']) // hex
    .replace(/!/g, dic['!']) // css !important
    .replace(/\//g, dic['/'])
    .replace(/\./g, dic['.'])
    .replace(/:/g, dic[':'])
    // https://github.com/sonofmagic/weapp-tailwindcss-webpack-plugin/issues/8
    .replace(/%/g, dic['%'])
    .replace(/,/g, dic[','])
    .replace(/\\/g, dic['\\'])
    .replace(/'/g, dic["'"])
    .replace(/"/g, dic['"'])
    .replace(/\*/g, dic['*'])
    .replace(/&/g, dic['&'])
    .replace(/@/g, dic['@'])
    .replace(/{/g, dic['{'])
    .replace(/}/g, dic['}'])
  if (!options.keepEOL) {
    res = res
      // 去除无用换行符和空格
      .replace(/[\r\n]+/g, '')
  }
  if (options.mangle) {
    res = mangleMark(res, options.classGenerator)
  }
  return res
}
