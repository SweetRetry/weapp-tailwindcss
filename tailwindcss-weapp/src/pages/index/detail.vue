<script setup lang="ts">
import BaseLayout from '@/components/BaseLayout.vue'
import Fab from '@/components/FloatButton.vue'
import Navbar from '@/components/Navbar.vue'
import { useSystemStore } from '@/stores'
import camelCase from 'lodash/camelCase'
import redent from 'redent'

const store = useSystemStore()

interface TableItem {
  Class: string
  Properties: string[]
}

const listData = ref<TableItem[]>([])
const tipsContent = `${['长按行可以复制当前行的值']
  .map((x, idx) => {
    return `${idx + 1}. ${x}`
  })
  .join('\n')}`

const plugin = ref('')
const title = ref('')
const tipVisible = ref(false)
const header = [
  {
    title: 'Class',
    key: 'Class',
  },
  {
    title: 'Properties',
    key: 'Properties',
  },
]

const currentUrl = computed(() => {
  return `/pages/index/detail?t=${title.value}`
})

function showTips() {
  tipVisible.value = true
}

function longpressCopy(item: TableItem) {
  const copyValue = [
    `${item.Class} {`,
    `  ${item.Properties.join('\n  ')}`,
    '}',
  ].join('\n')

  uni.setClipboardData({
    data: copyValue,
  })
}

onLoad((params) => {
  title.value = params?.t as string
  plugin.value = camelCase(title.value)
  switch (plugin.value) {
    case 'topRightBottomLeft': {
      plugin.value = 'inset'
      break
    }
    case 'screenReaders': {
      plugin.value = 'accessibility'
      break
    }
    default: {
      break
    }
  }
  if (plugin.value === 'animation') {
    listData.value = [
      {
        Class: 'animate-none',
        Properties: ['animation: none;'],
      },
      {
        Class: 'animate-spin',
        Properties: [
          redent(`
              animation: spin 1s linear infinite;\n
              @keyframes spin {
                from {
                  transform: rotate(0deg);
                }
                to {
                  transform: rotate(360deg);
                }
              }
            `).trim(),
        ],
      },
      {
        Class: 'animate-ping',
        Properties: [
          redent(`
              animation: ping 1s cubic-bezier(0, 0, 0.2, 1) infinite;\n
              @keyframes ping {
                75%, 100% {
                  transform: scale(2);
                  opacity: 0;
                }
              }
            `).trim(),
        ],
      },
      {
        Class: 'animate-pulse',
        Properties: [
          redent(`
              animation: pulse 2s cubic-bezier(0.4, 0, 0.6, 1) infinite;\n
              @keyframes pulse {
                0%, 100% {
                  opacity: 1;
                }
                50% {
                  opacity: .5;
                }
              }
            `).trim(),
        ],
      },
      {
        Class: 'animate-bounce',
        Properties: [
          redent(`
              animation: bounce 1s infinite;\n
              @keyframes bounce {
                0%, 100% {
                  transform: translateY(-25%);
                  animation-timing-function: cubic-bezier(0.8, 0, 1, 1);
                }
                50% {
                  transform: translateY(0);
                  animation-timing-function: cubic-bezier(0, 0, 0.2, 1);
                }
              }
            `).trim(),
        ],
      },
    ]
  }
  else {
    listData.value = Object.entries(store.getPluginsById(plugin.value)).map(
      (x) => {
        return {
          Class: x[0],
          // @ts-ignore
          Properties: Object.entries(x[1]).map(([key, value]) => {
            return `${key}: ${value};`
          }),
        }
      },
    )
  }

  uni.setNavigationBarTitle({
    title: title.value,
  })
})

onShareAppMessage(() => {
  return {
    title: plugin.value,
    path: currentUrl.value,
  }
})

onShareTimeline(() => {
  return {
    title: plugin.value,
    path: currentUrl.value,
  }
})
</script>

<template>
  <BaseLayout>
    <Navbar auto-back>
      <template #center>
        {{ title }}
      </template>
    </Navbar>
    <view class="px-4">
      <view class="tw-table">
        <view class="thead sticky top-0 z-50 bg-white transition-colors duration-500 dark:bg-slate-900">
          <view class="tr">
            <view class="w-4/12">
              <view class="flex py-2 pr-2 text-sm font-semibold text-slate-700 dark:text-slate-300">
                <text class="mr-2">
                  {{ header[0].title }}
                </text>
                <u-icon name="question-circle" @click="showTips" />
              </view>
            </view>
            <view class="w-8/12">
              <view class="py-2 pl-2 text-sm font-semibold text-slate-700 dark:text-slate-300">
                {{
                  header[1].title
                }}
              </view>
            </view>
          </view>
        </view>
        <view class="tbody">
          <view
            v-for="(item, index) in listData" :key="index" class="tr" hover-class="hover-gray-100"
            @longpress="longpressCopy(item)"
          >
            <template v-for="(obj, idx) in header" :key="idx">
              <view :class="idx === 0 ? 'w-4/12' : 'w-8/12'">
                <template v-if="Array.isArray(item[obj.key])">
                  <view class="whitespace-pre-wrap py-2 pl-2  text-xs text-indigo-600 dark:text-indigo-300">
                    <view v-for="(r, i) in item[obj.key]" :key="i">
                      {{ r }}
                    </view>
                  </view>
                </template>
                <template v-else>
                  <view
                    class="
                    whitespace-pre-wrap
                    py-2
                    pr-2 text-xs font-medium
                    text-sky-500
                    dark:text-sky-400
                  "
                  >
                    {{ item[obj.key] }}
                  </view>
                </template>
              </view>
            </template>
          </view>
        </view>
        <!-- <view></view> -->
      </view>

      <u-modal :show="tipVisible" title="操作提示" :content="tipsContent" @confirm="tipVisible = false" />
      <Fab store-key="detail-float-btn">
        <button
          class="
          u-reset-button
          pointer-events-auto
          flex
          size-10
          items-center
          justify-center
          rounded-full
          bg-white
          shadow-sm
          dark:bg-slate-700

        " open-type="share"
        >
          <i class="i-ri-wechat-fill text-xl text-[#07c160]" />
        </button>
      </Fab>
    </view>
  </BaseLayout>
</template>

<style lang="scss">
.tw-table {
  .thead {
    .tr {
      @apply border-b border-slate-200 dark:border-slate-400/20;
    }
  }

  .tbody {
    .tr {
      @apply border-b border-slate-100 dark:border-slate-400/10;
    }
  }

  .tr {
    @apply flex;

    .td {
    }
  }
}
</style>
