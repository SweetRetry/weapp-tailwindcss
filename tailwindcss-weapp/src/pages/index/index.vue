<script setup lang="ts">
import { devDependencies } from '../../../package.json'
import BaseLayout from '@/components/BaseLayout.vue'
import FloatButton from '@/components/FloatButton.vue'
import { useSystemStore } from '@/stores'

const version = devDependencies.tailwindcss.slice(1)

const store = useSystemStore()

function openSearchBox() {

}

function toggleTheme() {
  if (store.theme === 'dark') {
    store.setTheme('light')
  }
  else {
    store.setTheme('dark')
  }
}

function go2ThemeDemo() {
  uni.navigateTo({
    url: '/pages/theme/index',
  })
}
</script>

<template>
  <BaseLayout>
    <up-navbar fixed bgColor="inherit" safeAreaInsetTop>
      <template #left>
        <view class="relative flex items-center">
          <view class="i-logos-tailwindcss-icon mr-2 h-[20.57px] w-[34px] shrink-0" />
          <view class="text-lg font-semibold">
            Tailwind CSS
          </view>
          <view
            class="relative bottom-0.5 ml-1 self-end rounded-full bg-sky-400/10 px-2 py-0.5 text-xs font-medium text-sky-600"
          >
            {{ version }}
          </view>
        </view>
      </template>
    </up-navbar>
    <FloatButton store-key="index-float-btn" :padding="[256, 32, 256, 32]">
      <view class="float-btn mb-3">
        <view class="i-mdi-flask-round-bottom-empty-outline text-sky-400 dark:text-sky-500" @click="go2ThemeDemo" />
      </view>
      <view class="float-btn mb-3" @click="toggleTheme">
        <view v-if="store.theme === 'dark'" class="i-mdi-white-balance-sunny text-sky-500" />
        <view v-else class="i-mdi-weather-night text-sky-400" />
      </view>
      <view class="float-btn" @click="openSearchBox">
        <view class="i-mdi-magnify text-sky-400 dark:text-sky-500" />
      </view>
    </FloatButton>
  </BaseLayout>
</template>
