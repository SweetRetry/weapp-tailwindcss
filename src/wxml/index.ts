import { parse, render } from 'wxml-ast'
import { parseExpression } from '@babel/parser'
import traverse from '@babel/traverse'
import generate from '@babel/generator'
import { replaceWxml } from './shared'

export { replaceWxml }

// #region internal
const variableRegExp = /{{([^{}]*)}}/g

function variableMatch(original: string) {
  return variableRegExp.exec(original)
}
// #endregion

export function generateCode(match: string) {
  const ast = parseExpression(match)

  traverse(ast, {
    StringLiteral(path) {
      path.node.value = replaceWxml(path.node.value)
    },
    noScope: true
  })

  const { code } = generate(ast, {
    compact: true,
    minified: true
  })
  return code
}

export interface RawSource {
  start: number
  end: number
  raw: string
  // '' 直接 remove {{}}
  source?: string
}

export function templeteReplacer(original: string) {
  let match = variableMatch(original)
  const sources: RawSource[] = []

  while (match !== null) {
    // 过滤空字符串
    // if (match[1].trim().length) {
    sources.push({
      start: match.index,
      end: variableRegExp.lastIndex,
      raw: match[1]
    })

    match = variableMatch(original)
  }

  if (sources.length) {
    const resultArray: string[] = []
    let p = 0
    for (let i = 0; i < sources.length; i++) {
      const m = sources[i]
      // 匹配前值
      resultArray.push(replaceWxml(original.slice(p, m.start), true))
      p = m.start
      // 匹配后值
      if (m.raw.trim().length) {
        const code = generateCode(m.raw)
        m.source = `{{${code}}}`
      } else {
        m.source = ''
      }

      resultArray.push(m.source)
      p = m.end
      // 匹配最终尾部值
      if (i === sources.length - 1) {
        resultArray.push(replaceWxml(original.slice(m.end), true))
      }
    }

    return resultArray.filter((x) => x).join('')
  } else {
    return replaceWxml(original)
  }
}

export async function templeteHandler(rawSource: string) {
  const parsed = await parse(rawSource, {
    elementCB: (node) => {
      if (node.attribs.class) {
        node.attribs.class = templeteReplacer(node.attribs.class)
      }
    }
  })
  return render(parsed)
  // wxml.traverse(parsed, (node, parent) => {
  //   if (node.type === wxml.NODE_TYPES.ELEMENT) {
  //     // @ts-ignore
  //     if (node.attributes.class) {
  //       // @ts-ignore
  //       node.attributes.class = templeteReplacer(node.attributes.class)
  //     }
  //   }
  // })
  // return wxml.serialize(parsed)
}
