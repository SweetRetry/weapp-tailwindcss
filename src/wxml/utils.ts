import * as t from '@babel/types'
import { replaceWxml } from './shared'
import { parseExpression, traverse, generate } from '@/babel'
import { variableMatch, variableRegExp, templateClassExactRegexp, tagWithEitherClassAndHoverClassRegexp, makeCustomAttributes } from '@/reg'
import { defu } from '@/utils'
import type { RawSource, ICommonReplaceOptions, ITempleteHandlerOptions } from '@/types'

export function generateCode(match: string, options: ICommonReplaceOptions = {}) {
  const ast = parseExpression(match)

  traverse(ast, {
    StringLiteral(path) {
      // [g['人生']==='你好啊'?'highlight':'']
      if (t.isMemberExpression(path.parent)) {
        return
      }
      // parentPath maybe null
      // ['td',[(g.type==='你好啊')?'highlight':'']]
      if (t.isBinaryExpression(path.parent) && t.isConditionalExpression(path.parentPath?.parent)) {
        return
      }

      path.node.value = replaceWxml(path.node.value, options)
    },
    noScope: true
  })
  // @ts-ignore
  const { code } = generate(ast, {
    compact: true,
    minified: true,
    jsescOption: {
      quotes: 'single',
      minimal: true
    }
  })

  return code
}

export function extractSource(original: string) {
  let match = variableMatch(original)
  const sources: RawSource[] = []

  while (match !== null) {
    // 过滤空字符串
    // if (match[1].trim().length) {
    const start = match.index
    const end = variableRegExp.lastIndex
    sources.push({
      start,
      end,
      raw: match[1],
      prevConcatenated: !/\s/.test(original[start - 1]),
      nextConcatenated: !/\s/.test(original[end])
    })

    match = variableMatch(original)
  }
  return sources
}

export function templeteReplacer(original: string, options: ICommonReplaceOptions = {}) {
  const sources = extractSource(original)

  if (sources.length > 0) {
    const resultArray: string[] = []
    let p = 0
    for (let i = 0; i < sources.length; i++) {
      const m = sources[i]
      const before = original.slice(p, m.start)

      // 匹配前值
      resultArray.push(
        replaceWxml(before, {
          keepEOL: true,
          escapeMap: options.escapeMap
        })
      )
      p = m.start
      // 匹配后值
      if (m.raw.trim().length > 0) {
        const code = generateCode(m.raw, options)
        const source = `{{${code}}}`
        m.source = source
      } else {
        m.source = ''
      }

      resultArray.push(m.source)
      p = m.end
      // 匹配最终尾部值
      if (i === sources.length - 1) {
        const after = original.slice(m.end)
        resultArray.push(
          replaceWxml(after, {
            keepEOL: true,
            escapeMap: options.escapeMap
          })
        )
      }
    }

    return resultArray
      .filter(Boolean)
      .join('')
      .trim()
  } else {
    return replaceWxml(original, {
      keepEOL: false,
      escapeMap: options.escapeMap
    })
  }
}

export function templeteHandler(rawSource: string, options: ICommonReplaceOptions = {}) {
  return rawSource.replace(tagWithEitherClassAndHoverClassRegexp, (m0) => {
    return m0.replace(templateClassExactRegexp, (m1, className) => {
      return m1.replace(className, templeteReplacer(className, options))
    })
  })
}

export function customTempleteHandler(rawSource: string, options: ITempleteHandlerOptions = {}) {
  let source = templeteHandler(rawSource, options)
  const regexps = makeCustomAttributes(options.customAttributesEntities)
  if (regexps) {
    if (Array.isArray(regexps)) {
      for (const regexp of regexps) {
        source = source.replace(regexp.tagRegexp, (m0) => {
          return m0.replace(regexp.attrRegexp, (m1, className) => {
            return m1.replace(className, templeteReplacer(className, options))
          })
        })
      }
    }

    return source
  } else {
    return source
  }
}

export function createTempleteHandler(options: ITempleteHandlerOptions = {}) {
  return (rawSource: string, opt: ITempleteHandlerOptions = {}) => {
    return customTempleteHandler(rawSource, defu(opt, options))
  }
}
