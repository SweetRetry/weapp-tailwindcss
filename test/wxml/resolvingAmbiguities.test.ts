import { templeteHandler } from '@/wxml/index'
// import { format } from '../util'

describe('resolving ambiguities', () => {
  test('text-[22px]', () => {
    expect(templeteHandler('<div class="text-[22px]">...</div>')).toMatchSnapshot()
  })

  test('text-[#bada55]', () => {
    expect(templeteHandler('<div class="text-[#bada55]">...</div>')).toMatchSnapshot()
  })

  test('text-[var(--my-var)]', () => {
    expect(templeteHandler('<div class="text-[var(--my-var)]">...</div>')).toMatchSnapshot()
  })

  test('text-[length:var(--my-var)]', () => {
    expect(templeteHandler('<div class="text-[length:var(--my-var)]">...</div>')).toMatchSnapshot()
  })

  test('text-[color:var(--my-var)]', () => {
    expect(templeteHandler('<div class="text-[color:var(--my-var)]">...</div>')).toMatchSnapshot()
  })
})
