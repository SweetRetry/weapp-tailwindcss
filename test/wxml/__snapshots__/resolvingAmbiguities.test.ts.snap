// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolving ambiguities text-[#bada55] 1`] = `"<div class="text-_bl__h_bada55_br_">...</div>"`;

exports[`resolving ambiguities text-[22px] 1`] = `"<div class="text-_bl_22px_br_">...</div>"`;

exports[`resolving ambiguities text-[color:var(--my-var)] 1`] = `"<div class="text-_bl_color_c_var_pl_--my-var_qr__br_">...</div>"`;

exports[`resolving ambiguities text-[length:var(--my-var)] 1`] = `"<div class="text-_bl_length_c_var_pl_--my-var_qr__br_">...</div>"`;

exports[`resolving ambiguities text-[var(--my-var)] 1`] = `"<div class="text-_bl_var_pl_--my-var_qr__br_">...</div>"`;
